### PR TITLE
[MRG] Fix edf reader when preload=False.

### PR DESCRIPTION
**Pull request by jaeilepp**
Closes https://github.com/mne-tools/mne-python/issues/3894

In the end, the solution was really simple. We should probably update the test files to have channels with different sampling rates.

### CODE REVIEW

**Comment by agramfort** [comment]
+1 for a test if you can find a tiny one.

cc @slasnista

File: doc/whats_new.rst [hunk 1]
```diff
@@ -106,6 +106,7 @@ BUG
 
     - Raise error if the cv parameter of :class:`mne.decoding.GeneralizationAcrossTime` and :class:`mne.decoding.TimeDecoding` is not a partition and the predict_mode is "cross-validation" by `Jean-Remi King`_
 
+    - Fix bug in :func:`mne.io.read_raw_edf` when `preload=False` and channels have different sampling rates by `Jaakko Leppakangas`_
```
Comment (Eric89GXL) [inline doc/whats_new.rst]: double-backticks on `preload=False`

**Comment by jaeilepp** [comment]
I was able to modify the test file with edfbrowser. The new file is the same as the old one, but some of the channels are resampled to various frequencies. Maybe we can even get rid of the old test file.

**Comment by agramfort** [comment]
how long does the test take?

how big is the file?

does the test fail on master?

**Comment by jaeilepp** [comment]
The whole test_edf module takes 1 sec for me. It's basically the same file as before, but some of the channels are downsampled so it's a bit smaller (820kB) and yes, the test fails on master.

**Comment by agramfort** [comment]
perfect !

+1 for merge then.

**Comment by Eric89GXL** [comment]
I don't like adding files to the repo for testing anymore. It's not sustainable from a repo bloat perspective. Can we add the file to `mne-testing-data` instead?

**Comment by jaeilepp** [comment]
How about I just remove the old file? It's basically the same data. I think the clean up should be on it's own PR.

**Comment by agramfort** [comment]
if you remove the old file it will remain in the commit history and will
not reduce the repo size.

 mne-testing-data

is certainly the cleanest option.

**Comment by jaeilepp** [comment]
Ok. testing-data updated.

**Comment by agramfort** [comment]
thx @jaeilepp

**Pull request merged**